### ISSUE TITLE
Report an error for duplicate import symbols from different sources

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#4086]
**Is your feature request related to a problem? Please describe.**

I have encountered code that incorrectly imports the same symbol from two different places, e.g.

```py
# a.py
class Path:
    ...

# b.py
from pathlib import Path
from a import Path

def foo(path: Path) -> None:
    ...
```

This would have lead to bugs in our type definitions if mypy had not caught this error early on.

**Describe the solution you'd like**

Pyright should report an error when the same symbol is imported from two different places.

**Additional context**

Mypy reports an error for this code:

```
b.py:2: error: Incompatible import of "Path" (imported name has type "Type[a.Path]", local name has type "Type[pathlib.Path]")
```

**Comment by erictraut** [comment]
Unlikely mypy, which conflates certain assignments with implied declarations, pyright is consistent in differentiating between the two. If a symbols has no type declaration, any value can be assigned to it. For example, this is allowed in pyright:

```python
a = 1
a = "hi" # Mypy emits error here, pyright does not
```

But this is not:
```python
a: int = 1
a = "hi"
```

Pyright treats an import as an assignment to the symbol. It's not a type declaration. So the following is allowed:

```python
Path = 1
from pathlib import Path
Path = "hi"
```

But the following is not because it involves a type declaration and an incompatible assignment.
```python
class Path: ...
from pathlib import Path
```

If we were to implement this check in pyright, I think we'd want to place it under a separate diagnostic rule since the normal rules (as described above) wouldn't apply — or would need to be applied inconsistently.

I haven't looked at mypy's implementation of this check, but in playing around with some examples, it looks like it does not emit this check in cases where the symbol has a declared type.

```python
Path: type
from pathlib import Path
from a import Path # No Error
```

It also doesn't emit an error for duplicate imports if they resolve to the same type.

```python
from pathlib import Path
from pathlib import Path # No Error
```
It does emit an error if the imports are on different branches of a conditional.

```python
if some_condition:
    from pathlib import Path
else:
    from a import Path # Error
```

**Comment by erictraut** [comment]
I thought about this a bit more and discussed with a colleague on the pylance team. I'm not convinced this is a sufficiently common source of bug to justify adding a new diagnostic rule for it. Depending on the design of the check, it could lead to false positives, so it's not clear that we could include such a rule in the "basic" rule set.

I'm going to close this for now. We might reconsider it in the future if there's additional signal from pyright or pylance users.
